src/scan.js

```diff
@@ -21,7 +23,12 @@ var _curry3 = require('./internal/_curry3');
  *      var factorials = R.scan(R.multiply, 1, numbers); //=> [1, 1, 2, 6, 24]
  */
 module.exports = _curry3(function scan(fn, acc, list) {
-  var idx = 0, len = list.length, result = [acc];
+  if (!_isArray(list) && typeof list.scan === 'function') {
```

Comment:
I'd prefer to remove the guard, in that case. It strikes me as an unnecessary exception to this rule:

> Dispatches to the `scan` method of the third argument, if present.